shared-libs/transitions/test/unit/pregnancy_registration.js

```diff
@@ -401,3 +401,373 @@ describe('pregnancy registration', () => {
   });
 
 });
+
+describe('pregnancy registration with exact LMP date', () => {
+  const eightWeeksAgo = moment().subtract(8, 'weeks').startOf('day');
+  afterEach(() => sinon.restore());
+
+  beforeEach(() => {
+    transition.setExpectedBirthDate = transition.__get__('setExpectedBirthDate');
+    sinon.stub(config, 'get').returns([{
+      form: 'l',
+      type: 'pregnancy',
+      events: [
+        {
+          name: 'on_create',
+          trigger: 'add_patient',
+          params: '',
+          bool_expr: ''
+        },
+        {
+          name: 'on_create',
+          trigger: 'add_expected_date',
+          params: '',
+          bool_expr: 'typeof doc.getid === "undefined"'
+        }
+      ],
+      validations: {
+        join_responses: true,
+        list: [          
+          {
+            property: 'lmpDate',
+            rule: 'isAfter("-40 weeks")',
+            message: [{
+              content: 'Date should be later than 40 weeks ago.',
+              locale: 'en'
+            }]
+          },          
+          {
+            property: 'lmpDate',
+            rule: 'isBefore("8 weeks")',
+            message: [{
+              content: 'Date should be older than 8 weeks ago.',
+              locale: 'en'
+            }]
+          },
+          {
+            property: 'patient_name',
+            rule: 'lenMin(1) && lenMax(100)',
+            message: [{
+              content: 'Invalid patient name.',
+              locale: 'en'
+            }]
+          }
+        ]
+      }
+    }, {
+      // Pregnancy for existing patient
+      form: 'ep',
+      type: 'pregnancy',
+      events: [
+        {
+          name: 'on_create',
+          trigger: 'add_expected_date',
+          params: '',
+          bool_expr: 'typeof doc.getid === "undefined"'
+        }
+      ],
+      validations: {
+        join_responses: true,
+        list: [
+          {
+            property: 'patient_id',
+            rule: 'len(5)',
+            message: [{
+              content: 'Invalid patient Id.',
+              locale: 'en'
+            }]
+          }
+        ]
+      }
+    }]);
+  });
+
+  it('filter fails with empty doc', () => {
```

Comment:
These look like they are the same tests as in the other describe block. 
I don't think it's worth to duplicate these tests, as most of these don't even reach the changed code. 

I think it would be better to add a another `describe` block in the block above and just add the tests around the changed code, with explicit LMP date.

Reply:
Removed duplicate tests.